### PR TITLE
Prevent adding duplicate links

### PR DESCRIPTION
**Pull request by max-programming**
In this PR, I made the changes so that a link can't be shared multiple times. Let me know if it works fine

### CODE REVIEW

File: src/db/shares.ts [hunk 1]
```diff
@@ -83,19 +83,29 @@ export const markShareAsEmailed = async (id: string): Promise<Share> => {
 
 export const createShare = async (
   share: ShareInsert
-): Promise<ShareWithUsername> => {
-  const res = await supabase
+): Promise<ShareWithUsername | null> => {
+  const shared = await supabase
```
Comment (jamesqquick) [inline src/db/shares.ts:87]: Can we go ahead and destructure the data property originally? So:

`const {data} = await supabase...`

File: src/db/shares.ts [hunk 2]
```diff
@@ -83,19 +83,29 @@ export const markShareAsEmailed = async (id: string): Promise<Share> => {
...
+      );
+    if (res.error) {
+      throw res.error;
+    }
+
+    return res.data[0] as ShareWithUsername;
   }
 
-  return res.data[0] as ShareWithUsername;
+  return null;
```
Comment (jamesqquick) [inline src/db/shares.ts:108]: Can we add a console.info() statement here for debugging purposes?

File: src/commands/share.ts [hunk 3]
```diff
@@ -87,6 +87,11 @@ const shareHandler = async (
       emailable: false,
       emailed: false,
     });
+    if (!createdShare || createdShare === null) {
+      return interaction.reply({
+        content: 'This link is already shared',
```
Comment (jamesqquick) [inline src/commands/share.ts:92]: Can we update the tet to past tense? "This link was already shared"

**Review by jamesqquick — CHANGES_REQUESTED** [review]

File: src/db/shares.ts (hunk 1 above)
Comment (max-programming) [reply src/db/shares.ts]: Done ✌

**Review by jamesqquick — APPROVED** [review]
Thank you!

**Pull request merged**